samples/asymmetricEncrypt.js

```diff
@@ -0,0 +1,65 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+'use strict';
+
+// [START kms_asymmetric_encrypt]
+async function asymmetricEncrypt(
+  projectId = 'your-project-id', // Your GCP projectId
+  keyRingId = 'my-key-ring', // Name of the crypto key's key ring
+  cryptoKeyId = 'my-key', // Name of the crypto key, e.g. "my-key"
+  cryptoKeyVersionId = '1', // Version of the crypto key to use
+  plaintext = 'my data to encrypt' // Plaintext data to encrypt
+) {
+  // Import the library and create a client
+  const kms = require('@google-cloud/kms');
+  const client = new kms.KeyManagementServiceClient();
+
+  // The location of the crypto key's key ring, e.g. "global"
+  const locationId = 'global';
+
+  // Construct the cyrpto key version ID
+  const name = client.cryptoKeyVersionPath(
+    projectId,
+    locationId,
+    keyRingId,
+    cryptoKeyId,
+    cryptoKeyVersionId
+  );
+
+  // Get public key from Cloud KMS
+  const [publicKey] = await client.getPublicKey({ name: name });
+
+  // Encrypt plaintext locally using the public key. This example uses a key
+  // that was configured with sha256 hash with OAEP padding. Update these values
+  // to match the Cloud KMS key.
+  const crypto = require('crypto');
+  const plaintextBuffer = Buffer.from(plaintext);
+  const encryptedBuffer = crypto.publicEncrypt(
```

Comment:
was this API surface relatively similar to the other languages you looked at, I'm curious.

Reply:
Yup! I based this off of both the python and go examples. I think this is why people are requesting the samples. This really isn't a "Cloud KMS" example per se, but more "how the heck to do I use [lang] stdlib to do public key encryption?" Nonetheless, I think it's our responsibility to provide these samples, since it's critical to the journey of this product feature.